bm-persona/Common/CollapsibleCardView.swift

```diff
@@ -0,0 +1,146 @@
+        containerView.rightAnchor.constraint(equalTo: self.layoutMarginsGuide.rightAnchor).isActive = true
+        containerView.topAnchor.constraint(equalTo: self.layoutMarginsGuide.topAnchor).isActive = true
+        containerView.bottomAnchor.constraint(equalTo: self.layoutMarginsGuide.bottomAnchor).isActive = true
+        
+        containerView.addSubview(chevronIcon)
+        containerView.addSubview(collapsedView)
+        containerView.addSubview(openedView)
+        
+        chevronIcon.widthAnchor.constraint(equalToConstant: 16).isActive = true
+        chevronIcon.heightAnchor.constraint(equalToConstant: 16).isActive = true
+        chevronIcon.rightAnchor.constraint(equalTo: containerView.rightAnchor).isActive = true
+        chevronIcon.centerYAnchor.constraint(equalTo: collapsedView.centerYAnchor).isActive = true
+        
+        // constrain the collapsedView to icon or the left of the card if no icon
+        var leftConstraint = collapsedView.leftAnchor.constraint(equalTo: self.layoutMarginsGuide.leftAnchor)
+        if let leftIcon = leftIcon {
+            self.addSubview(leftIcon)
+            leftIcon.widthAnchor.constraint(equalToConstant: 16).isActive = true
+            leftIcon.heightAnchor.constraint(equalToConstant: 16).isActive = true
+            leftIcon.leftAnchor.constraint(equalTo: containerView.leftAnchor).isActive = true
+            leftIcon.centerYAnchor.constraint(equalTo: collapsedView.centerYAnchor).isActive = true
+            leftConstraint = collapsedView.leftAnchor.constraint(equalTo: leftIcon.rightAnchor, constant: kViewMargin)
+        }
+        
+        collapsedView.topAnchor.constraint(equalTo: containerView.topAnchor).isActive = true
+        collapsedView.rightAnchor.constraint(equalTo: chevronIcon.leftAnchor, constant: -1 * kViewMargin).isActive = true
+        collapsedView.heightAnchor.constraint(greaterThanOrEqualTo: chevronIcon.heightAnchor).isActive = true
+        leftConstraint.isActive = true
+        
+        openedView.leftAnchor.constraint(equalTo: collapsedView.leftAnchor).isActive = true
+        openedView.topAnchor.constraint(equalTo: collapsedView.bottomAnchor, constant: kViewMargin).isActive = true
+        openedView.rightAnchor.constraint(equalTo: collapsedView.rightAnchor).isActive = true
+        
+        self.layoutIfNeeded()
+        setState(opened: isOpen)
+    }
+    
+    // rotate the chevron depending on the opened state (point left or down)
+    func toggleState(completion: (() -> Void)? = nil) {
+        UIView.animate(withDuration: 0.2) {
+            self.setState(opened: !self.isOpen)
+            if self.isOpen {
+                self.chevronIcon.transform = self.chevronIcon.transform.rotated(by: -1 * .pi / 2)
+            } else {
+                self.chevronIcon.transform = self.chevronIcon.transform.rotated(by: .pi / 2)
+            }
+        } completion: { _ in
+            completion?()
+        }
```

Comment:
wasn't getting the error because I was using Xcode 12 beta and I think that comes with a new version of swift where this syntax is allowed. changed it to work in Xcode 11 as well